File: 1 basic/1008. 数组元素循环右移问题 (20).cpp
#include<cstdio>

int a[105]; 

// 翻转数组中在区间[m,n)之间的元素 
int reverse(int m,int n){
	for(int i=0,mid=(n-m)/2; i<mid;++i){
		int t = a[m+i];
		a[m+i] = a[n-i-1];
		a[n-i-1] = t;
	}
}


int main(){
	int n,m;
	scanf("%d%d",&n,&m);
	for(int i=0;i<n;++i)
		scanf("%d",a+i);
		
	m %= n;
	reverse(0,n-m);
	reverse(n-m,n);
	reverse(0,n);
	
	printf("%d",a[0]);
	for(int i=1;i<n;++i)
		printf(" %d",a[i]);
	
	return 0;
}

File: 1 basic/py/1008. 数组元素循环右移问题 (20).py
def reverse(a,m,n):
    mid = int((n - m) / 2)
    for i in range(mid):
        t = a[m + i ]
        a[m + i ]= a[n - i -1]
        a[n - i -1] = t 

s = input().split()
n,m = int(s[0]),int(s[1])
a = input().split()
a = [int(x) for x in a ]
m = m % n
reverse(a, 0, n - m)
reverse(a, n - m, n)
reverse(a, 0, n)
ans = ''
for x in a: ans = ans + str(x)+ ' '
print(ans.strip())